server/src/main/scala/de/frosner/broccoli/templates/CachedTemplateSource.scala

```diff
@@ -0,0 +1,24 @@
+package de.frosner.broccoli.templates
+
+import de.frosner.broccoli.models.Template
+import javax.inject.Singleton
+
+/**
+  * The template source that wraps another template source and caches loaded templates of that source
+  *
+  * @param source The template source that will be cached
+  */
+@Singleton
```

Comment:
Please remove this annotation; it's not used since the source is created through a provider method which has its own `@Singleton` annotation.